### ISSUE TITLE
Assembly: Failing if assembly placement is modified

### ISSUE THREAD

**Issue by PaddleStroke** [FreeCAD/FreeCAD#17398]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

If you modify the placement of an assembly, say you rotate it a little bit, then many things go wrong : 
Joints are not finding the correct placement.
Assembly dragger fails to move objects correctly.
Probably more issues.

Perhaps we should hide the Placement property of AssemblyObjects. And also we could prevent the use of transform on the assembly. 
Thoughts?

### Full version info

```shell
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 1.1.0dev.39026 +5 (Git)
Build type: Release
Branch: core_associate_view
Hash: 50babf4d32fa83a1e3ad099b85ef0207cb883c27
Python 3.8.10, Qt 5.15.2, Coin 4.0.1, Vtk , OCC 7.6.3
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: OpenDark.qss/FreeCAD Classic/Qt default
Installed mods: 
  * A2plus 0.4.60k
  * Assembly4 0.50.2
  * BillOfMaterials-WB-main 0.0.8.2
  * Curves 0.6.8
  * ExtremeProLight 1.5.0
  * fasteners 0.5.0
  * FreeCAD - Shortcut.lnk
  * freecad.gears 1.2.0
  * FreeCAD_FastenersWB-master 0.4.56
  * kicadStepUpMod 10.23.0
  * Ondsel 2024.9.13.02
  * OpenDark 2023.12.17
  * ProDarkThemePreferencePack 1.0.0
  * sheetmetal 0.2.63
```

### Subproject(s) affected?

Assembly

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
So does it fail also when I create an assembly and then use this as a sub assembly in a different or the same file? Or is it just failing when the placement property of the root level assembly of a file is altered?

**Comment by pierreporte** [comment]
> Since there're no real usecase to move an assembly.

This is only true for an assembly that happen to be the root of the document.

**Comment by tomster** [comment]
I can definitely confirm this report.In particular, once the assembly has been moved (via Transform) the indicators of the joint connectors i.e. when creating a fixed joint seem to be offset by the distance that the assembly has traveled thus making it very difficult and sometimes even impossible to create a working joint.

I've added a minimal assembly that demonstrates this behavior. I.e. try constraining the top of the box with a fixed joint and you can see what I've described. If you reset the placement of the assembly to [0,0,0] everything works again, as expected.

Hope this helps.

[Issue17398.zip](https://github.com/user-attachments/files/17689094/Issue17398.zip)

**Comment by FEA-eng** [comment]
@PaddleStroke Can you confirm if this issue is still relevant?

**Comment by PaddleStroke** [comment]
yes

**Comment by luzpaz** [comment]
Any further discussion ?

**Comment by PaddleStroke** [comment]
No. This problem is not  very critical anyway, user should not be transforming the assemblies. Well some might nest assembly objects, which is not ideal but doable. Then they may have this issue while editing the nested assembly.

Anyway this needs to wait for https://github.com/FreeCAD/FreeCAD/pull/25513 before any further advances.

**Comment by tomster** [comment]
> No. This problem is not very critical anyway, user should not be transforming the assemblies. 

I don't understand. Like at all. If i cannot (or should not) transform assemblies then I would naturally end up with a bunch of assemblies "sitting on top of each other". My workflow is to create an assembly, then move it out of the way, create the next and so on until I have all the parts i need and then assemble those to produce the final artefact (i.e. multiple boards are assembled to create the sides of a furniture, more parts are assembled to corner posts and finally all of them are assembled i.e. to a bed). 

what would be the alternative (or "correct") approach? I feel like I'm missing a fundamental concept.

I'd like to offer to collect input on this and write it up as a tutorial since the [official tutorial does not touch on these points](https://blog.freecad.org/2024/09/30/tutorial-getting-started-with-the-assembly-workbench/)

**Comment by pierreporte** [comment]
@tomster You are right. Moving assemblies (or sub-assemblies) around is necessary but maybe the issue is not clear enough to really describe the problem. As I wrote last year, transforming an assembly doesn’t make sense if it is the only root of a file.
